packages/lib-taquito/index.ts

```diff
@@ -0,0 +1,111 @@
+}
+
+type WalletMetadata = GetWalletResponse
+
+const assertSigned = (res: GenerateSignatureResponse) => {
+  if (res.status === 'Failed') {
+    throw new DfnsError(-1, 'signing failed', res)
+  } else if (res.status !== 'Signed') {
+    throw new DfnsError(
+      -1,
+      'cannot complete signing synchronously because this wallet action requires policy approval',
+      res
+    )
+  }
+}
+
+export class DfnsWallet implements Signer{
+  private options: Required<DfnsWalletOptions>
+  private readonly dfnsClient
+
+  private constructor(private metadata: WalletMetadata, options: DfnsWalletOptions, private tezosPubKey: string) {
+    this.dfnsClient = options.dfnsClient
+  }
+
+  public static async init(options: DfnsWalletOptions) {
+    const { walletId, dfnsClient } = options
+    const res = await dfnsClient.wallets.getWallet({ walletId })
+
+    if (res.status !== 'Active') {
+      throw new DfnsError(-1, 'wallet not active', { walletId, status: res.status })
+    }
+
+    if (res.network !== 'Tezos' && res.network !== 'TezosGhostnet') {
+      throw new DfnsError(-1, 'wallet is not bound to Tezos or TezosGhostnet', { walletId, network: res.network })
+    }
+
+    // get the public key encoded with the right tezos prefix.
+    const tezosPublicKey = this.getEncodedPublicKey(res.signingKey.publicKey, res.signingKey.scheme)
+
+    return new DfnsWallet(res, options, tezosPublicKey)
+  }
+
+  private static getEncodedPublicKey(publicKey: string, scheme: string): string {
+    switch (scheme) {
+      case 'EdDSA':
+        return b58cencode(publicKey, prefix[Prefix.EDPK])
+      case 'ECDSA':
+        return b58cencode(publicKey, prefix[Prefix.SPPK])
+      default:
+        throw new Error(`KeyScheme.${scheme} is not supported for tezos`)
+    }
+  }
+
+  public get address(): string {
+    // Tezos-bound wallets will have a tezos address
+    return this.metadata.address!
+  }
+
+  // publicKey implements the Taquito Signer interface
+  publicKey(): Promise<string> {
+    return Promise.resolve(this.tezosPubKey)
+  }
+
+  // publicKeyHash implements the Taquito Signer interface
+  publicKeyHash(): Promise<string> {
+    return Promise.resolve(this.metadata.address!)
+  }
+
+  // secretKey implements the Taquito Signer interface
+  secretKey(): Promise<string | undefined> {
+    throw undefined;
+  }
+
+  // sign implements the Taquito Signer interface
+  // For now, we only want to support transaction signing
+  async sign(op: string, magicByte?: Uint8Array | undefined): Promise<{ bytes: string; sig: string; prefixSig: string; sbytes: string; }> {
+    if (magicByte?.toString() !== TxWatermark.toString()) {
+      throw new DfnsError(-1, `magicBytes ${magicByte} not supported`, { walletId: this.metadata.id, network: this.metadata.network })
+    }
+    
+    
+    const res = await this.dfnsClient.wallets.generateSignature({
+      walletId: this.metadata.id,
+      body: { kind: 'Transaction', transaction: op },
```

Comment:
minor, I don't remember, but for consistency, if `op` is not prefixed with `0x`, we should add the prefix.

Reply:
I tested with or without 0x. It works. But I can add it